pandas/core/internals/blocks.py

```diff
@@ -2859,16 +2854,15 @@ def _extend_blocks(result, blocks=None):
     return blocks
 
 
-def _block_shape(values, ndim=1, shape=None):
+def _block_shape(values: ArrayLike, ndim: int = 1) -> ArrayLike:
     """ guarantee the shape of the values to be at least 1 d """
     if values.ndim < ndim:
-        if shape is None:
-            shape = values.shape
-        if not is_extension_array_dtype(values):
-            # TODO: https://github.com/pandas-dev/pandas/issues/23023
+        shape = values.shape
+        if not is_extension_array_dtype(values.dtype):
```

Comment:
is this needed? (as you are calling the check in the block constructor).

Reply:
yes, we get a bunch of test failures without this check

Reply:
hmm, i think we DO want to remove this from inside block_shape though, no?

Reply:
this is the whole point of block_shape, im not clear on what you're suggesting

Reply:
you have a check IN the Block constructor AND in the _block_shape function itself.

Reply:
you should have 1 or the other

Reply:
but we dont have a _reshape_ call in the constructor

Reply:
obviously, i think we should have 2D EAs and not have to hassle with _any_ of this